src/ng1/angular1.ts

```diff
@@ -80,21 +83,41 @@ function ng1UIRouter($locationProvider) {
     };
 
     bindFunctions(["replace", "url", "path", "search", "hash"], $location, services.location);
-    bindFunctions([ 'port', 'protocol', 'host'], $location, services.locationConfig);
+    bindFunctions(['port', 'protocol', 'host'], $location, services.locationConfig);
     bindFunctions(['baseHref'], $browser, services.locationConfig);
 
     return router;
   }
 }
 
+function resolveProvider() {
+  this.$get = $get;
+  function $get() {
+    return {
+      resolve: (invocables, locals, parent, self) => {
+        let context = new ResolveContext([]);
```

Comment:
Give this a path with false `state`, i.e.:

```
let node = { state: {} }
let context = new ResolveContext([node]);
```

Then, you can add resolvables to that node like so:
```
    context.addResolvables(resolvables, node.state);
```